File: dart/lib/benchmark_result.dart
class BenchmarkResult {
  final String test;
  final int durationNs;
  final int memoryBytes;
  final int operations;
  final double opsPerSec;

  BenchmarkResult({
    required this.test,
    required this.durationNs,
    required this.memoryBytes,
    required this.operations,
    required this.opsPerSec,
  });

  Map<String, dynamic> toJson() => {
        'test': test,
        'duration_ns': durationNs,
        'memory_bytes': memoryBytes,
        'operations': operations,
        'ops_per_sec': opsPerSec,
      };
}

File: csharp/BenchmarkResult.cs
using System.Text.Json.Serialization;

namespace CSharpBenchmark;

public class BenchmarkResult
{
    [JsonPropertyName("test")]
    public string Test { get; set; } = string.Empty;
    
    [JsonPropertyName("duration_ns")]
    public long DurationNs { get; set; }
    
    [JsonPropertyName("memory_bytes")]
    public long MemoryBytes { get; set; }
    
    [JsonPropertyName("operations")]
    public long Operations { get; set; }
    
    [JsonPropertyName("ops_per_sec")]
    public double OpsPerSec { get; set; }

    public BenchmarkResult() { }

    public BenchmarkResult(string test, long durationNs, long memoryBytes, long operations, double opsPerSec)
    {
        Test = test;
        DurationNs = durationNs;
        MemoryBytes = memoryBytes;
        Operations = operations;
        OpsPerSec = opsPerSec;
    }
}